x/mongo/driver/operation/count.go

```diff
@@ -54,9 +55,12 @@ func buildCountResult(response bsoncore.Document) (CountResult, error) {
 	for _, element := range elements {
 		switch element.Key() {
 		case "n": // for count using original command
-			var ok bool
-			cr.N, ok = element.Value().AsInt64OK()
```

Comment:
Please revert this change. `AsInt64OK` already supports reading `Int32`, `Int64`, and `Double`.

Reply:
Great suggestion, will revert.

Reply:
Very true! My bad.

Reply:
Reverted.